docs/technical-doc/architecture.rst

```diff
@@ -0,0 +1,340 @@
+
+Note that operations stored in the Pool are naturally discarded after a certain time, since operations come with an expiration date in the `expiration_period` field. Still, some potential attacks can occur by trying to flood the pool with high fees operations that have no chance of being executed because the corresponding account does not have the required funds. Discussing about countermeasure for this is beyond the scope of this introduction.
+
+Block/Endorsement Factory Module
+********************************
+
+The Block Factory Module is in charge of creating new blocks when the corresponding node address has been designated to be the block creator for a given slot. This information is provided to the Factory Module from the Selector Module via the API Module.
+
+The Block Factory Module also needs information about the best parents (made of the latest blocks in each thread in the blockclique) from the Graph/Consensus Module. These parents will be included in the newly created block. Balance information, in order to assess the validity of pending operations, is obtained from the Execution Module, which maintains the ledger state from the point of view of the slot where the new block is supposed to be created.
+
+The Block Factory Module picks pending operations from the Pool Module. Note that the Block Factory will regularly query the Execution Module about finalized and executed operations, and internally cleanup operations that have been handled.
+
+Finally, the Block Factory will query the the Pool Module and pick pending endorsements corresponding to the best parents that are selected for the block.
+
+With this information, it is able to forge a new block that will then be propagated to the Graph/Consensus Module via the API Module, as well as to other nodes via gossip, to maintain a global synchronized state.
+
+the Endorsement Factory Module works in a similar manner, requesting the Selector Module to find out when it has been designated to be an endorsement producer, then feeding new endorsements to the Pool Module and the API Module for global synchronization.
+
+Operation lifecycle
+===================
+
+We have now all the elements and vocabulary in place to explore the lifecycle of an operation within the network, from creation to permanent execution in a finalized block.
+
+Operations can originate from two kinds of sources:
+
+1. Externally from a client that is forging an operation, for example: a transaction or a smart contract code execution. The client will have to know the IP address of a Massa Node (this can be either because it is a node itself and will simply use localhost, or via some maintained list of known nodes and/or some browser plugin), and will then send the operation to the API Module.
+2. Internally from a node that is executing code that triggers some operation, or via a mechanism of smart contract waking up event that allows for autonomous emission of operations (this feature is unique to Massa)
```

Comment:
What are the cases where this happens ? It's probably the case for endorsements if you count them as operations but otherwise it's mostly through API, either using the massa-client or through an external call. I may be wrong on this one.

Reply:
OK, I think I'm assuming something wrong here, which is that wake up call are made via operations, but in fact it's probably not the case. As you said, it's all implicit and does not need to go through the operation pipeline. So, in fact, the only actual operations that exist (=end up in pools) are operations externally produced by a client, right?

Reply:
Yes!